XenModel/Actions/Updates/DownloadAndUpdateClientAction.cs

```diff
@@ -210,7 +205,6 @@ protected override void Run()
             }
 
             Description = Messages.COMPLETED;
-            MarkCompleted();

```

Comment:
Calling MarkCompleted is not needed, because it is called by the base class after Run() has fininshed.

(I know this is done in another one or two places in the application, but in fact it is bad design of the base class. MarkCompleted should be private and not protected. But this fix is out of scope here).